File: services/chess_service/src/main/java/net/chess_platform/chess_service/chess/piece/Knight.java
package net.chess_platform.chess_service.chess.piece;

import java.util.List;

import net.chess_platform.chess_service.chess.Chessboard;
import net.chess_platform.chess_service.chess.move.Move;
import net.chess_platform.chess_service.chess.piece.behavior.KnightBehavior;
import net.chess_platform.chess_service.chess.piece.behavior.PieceBehavior;

public class Knight extends AbstractPiece {

    private final PieceBehavior knightMoves = new KnightBehavior();

    public Knight(Color color) {
        super(color, Type.KNIGHT);
    }

    @Override
    public List<Move> getMoves(Chessboard board, int row, int col) {
        return knightMoves.getMoves(board, getType(), getColor(), row, col);
    }

    @Override
    public String toString() {
        return getColor() == Color.WHITE ? Character.toString(9816) : Character.toString(9822);
    }
}

File: app/src/chess/pieces/Knight.ts
import { Chessboard } from "../Chessboard";
import { Move } from "../moves/Move";
import { AbstractPiece } from "./AbstractPiece";
import { KnightBehavior } from "./behavior/KnightBehavior";
import { Color } from "./Piece";

export class Knight extends AbstractPiece {

    private behavior = new KnightBehavior();

    constructor(color: Color, moveCount?: number) {
        super(color, "KNIGHT", moveCount);
    }

    getLegalMovesImpl(board: Chessboard, row: number, col: number): Move[] {
        return this.behavior.getMoves(board, this.getType(), this.getColor(), row, col);
    }

    override toString(): string {
        return this.getColor() == "WHITE" ? String.fromCodePoint(9816) : String.fromCodePoint(9822);
    }
}